app/forms.py

```diff
@@ -45,6 +51,19 @@ def clean_content_type(self):
         cleaned_data["content_type"] = ContentType.objects.get_for_model(self.object.__class__)
         return cleaned_data["content_type"]
 
+    def clean_content(self):
+        data = self.cleaned_data["content"]
+        if data.lower().startswith("mailto:"):
+            validate_email(data[7:])
+        elif data.lower().startswith("tel:"):
+            validate_tel = RegexValidator(r"^[0-9]+$", "Enter a valid telephone number")
+            validate_tel(data[4:])
+        else:
+            validate_url = URLValidator()
+            validate_url(data)
+        data = escape(data)
+        return data
+
```

Comment:
I figured it better to clean the data beforehand, for if a malicious link were to be submitted, the regex would catch it and prevent it from rendering, but would also yield a 500 to that page. Thus, this would be preventative of that situation.